### ISSUE TITLE
I get the error 'jl_atomic_load_relaxed: identifier not found.'

### ISSUE THREAD

**Issue by ytecer** [JuliaLang/julia#39595]
I am using Julia 1.6.0-rc1. As stated at https://docs.julialang.org/en/v1/manual/embedding/, I made the necessary settings in Visual Studio 2019, but when I compile the code, I get the error 'jl_atomic_load_relaxed: identifier not found.'

Code:
```c
#include <uv.h>
#include <Windows.h>
#include <julia.h>

int main(int argc, char* argv[])
{
    jl_init();
    jl_eval_string("print(sqrt(2.0))");
    jl_atexit_hook(0);
    return 0;
}
```
Error:
```
'jl_atomic_load_relaxed': identifier not found    julia.h     880
'jl_atomic_load_relaxed': identifier not found    julia.h     924
'jl_atomic_load_relaxed': identifier not found    locks.h   24
'jl_atomic_load_relaxed': identifier not found    locks.h   40
```

**Comment by Keno** [comment]
Looks like the MSVC versions of these are missing from atomics.h

**Comment by ytecer** [comment]
I think the following lines should be added to the atomics.h file.
```c
template<typename T>
static inline T jl_atomic_load_relaxed(volatile T *obj)
{
 #code....
 #code....
}
```

**Comment by balenamiaa** [comment]
Is there a reason compiler builtins are used instead of a higher level abstraction like stdatomic.h or mintomic?

**Comment by toivoh** [comment]
I got it to work with

    template<typename T>
    static inline T jl_atomic_load_relaxed(volatile T *obj)
    {
        return jl_atomic_load_acquire(obj);
    }

I guess this is not ideal, since `jl_atomic_load_acquire` should be stronger than `jl_atomic_load_relaxed`, but at least that means that is should work.

Some googling led me to believe that `InterlockedOrNoFence` might be used instead in MSVC, but I'm not sure enough about the details to try it myself.

**Comment by toivoh** [comment]
I also had to handle https://github.com/JuliaLang/julia/issues/34201 to get embedding to work (tried with Julia 1.6.0 in MSVC 2017), described what I did in that issue.

**Comment by vtjnash** [comment]
Duplicate of #42098, fixed by #42152
